gulpfile.js

```diff
@@ -0,0 +1,205 @@
+    }
+    if (!basePath.endsWith('/')) {
+      basePath = basePath + '/';
+    }
+    buildStream = buildStream.pipe(polymerProject.updateBaseTag(basePath));
+  }
+
+  if (options.addPushManifest) {
+    buildStream = buildStream.pipe(polymerProject.addPushManifest());
+  }
+
+  // If this option is specified, then we have to rename node_modules in our output
+  if (options.nodeModulesName) {
+    buildStream = buildStream
+      // Replace all the "node_modules" occurencies in files
+      .pipe(replace(/node_modules/g, options.nodeModulesName))
+      // Rename all the "node_modules" to the specified value
+      .pipe(rename((path) =>
+        path.dirname = path.dirname.replace(/node_modules/g, options.nodeModulesName)));
+  }
+
+  // Finish the build stream by piping it into the final build directory.
+  buildStream = buildStream.pipe(gulp.dest(buildDirectory));
+
+  // If a service worker was requested, parse the service worker config file
+  // while the build is in progress. Loading the config file during the build
+  // saves the user ~300ms vs. loading it afterwards.
+  const swPrecacheConfigPath = path.resolve(
+    polymerProject.config.root,
+    options.swPrecacheConfig || 'sw-precache-config.js');
+  let swConfig = null;
+  if (options.addServiceWorker) {
+    swConfig = require(swPrecacheConfigPath);
+  }
+
+  // There is nothing left to do, so wait for the build stream to complete.
+  await waitFor(buildStream);
+
+  if (options.addServiceWorker) {
+    await addServiceWorker({
+      buildRoot: buildDirectory,
+      project: polymerProject,
+      swPrecacheConfig: swConfig || undefined,
+      bundled: bundled,
+    });
+  }
+
+  logger.info(`${logPrefix}(${buildName}) Build complete!`);
+};
+
+/* ********** TASKS ******************************************************** */
+
+/**
+ * Cleans the static build directory (<project folder>/build)
+ */
+gulp.task('clean:static', () =>
+  del`build`);
+
+/**
+ * Cleans the PRPL server build directory (<project folder>/server/build)
+ */
+gulp.task('clean:prpl-server', () =>
+  del`server/build`);
+
+/**
+ * Builds a static version of the PWA that can be used on any hosting service
+ */
+gulp.task('build:static', () => {
+  const project = new PolymerProject(projectConfig);
+  return Promise.all(projectConfig.builds.map((buildConfig) =>
+    build({
+      ...buildConfig,
+      outputDir: 'build',
+      logPrefix: '[build:static] ',
+    }, project)));
+});
+
+/**
+ * Builds the PRPL-server-ready version of the PWA, auto setting the base path
+ * and renaming the node_modules folder, otherwise services like App Engine won't
+ * upload it
+ */
+gulp.task('build:prpl-server', () => {
+  const autoBasePathedConfig = {
+    ...projectConfig,
+    builds: projectConfig.builds.map((build) => ({
+      ...build,
+      basePath: true,
+    })),
+  };
+  const project = new PolymerProject(autoBasePathedConfig);
+  return Promise.all(autoBasePathedConfig.builds.map((buildConfig) =>
```

Comment:
Need to output polymer.json so that it can be used by `npm run serve:prpl-server` - similar to https://github.com/Polymer/tools/blob/0cdecd03e7e3709f84fa4f5a52443c19988112be/packages/cli/src/commands/build.ts#L214